pkg/api/api.go

```diff
@@ -60,11 +71,103 @@ func (a *API) routes() chi.Router {
 	r := chi.NewRouter()
 	r.Get("/nodes/all", a.nodes)
 	r.Get("/nodes/enabled", a.enabled)
+	r.Post("/nodes/specific/statements", a.specificNodesHandler)
 	return r
 }
 
+type NodeShortStatement struct {
+	Node    string `json:"node,omitempty"`
+	Version string `json:"version,omitempty"`
+	Height  int    `json:"height,omitempty"`
+}
+
+func DecodeValueFromBody(body io.ReadCloser, v interface{}) error {
+	if err := json.NewDecoder(body).Decode(v); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *API) specificNodesHandler(w http.ResponseWriter, r *http.Request) {
+	buf, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Failed to read body: %v", err)
+		http.Error(w, fmt.Sprintf("Failed to read body: %v", err), http.StatusInternalServerError)
+		return
+	}
+	stateHashReader := io.NopCloser(bytes.NewBuffer(buf))
```

Comment:
Please, add TODO to replace duplication after implementation of new exteneded state hash structure in gowaves.